ui/hooks/useAssetDetails.js

```diff
@@ -0,0 +1,103 @@
+import { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getCollectibles, getTokens } from '../ducks/metamask/metamask';
+import { ERC1155, ERC721, ERC20 } from '../helpers/constants/common';
+import {
+  calcTokenAmount,
+  getAssetDetails,
+  getTokenAddressParam,
+  getTokenValueParam,
+} from '../helpers/utils/token-util';
+import { parseTransactionData } from '../helpers/utils/transactions.util';
+import { getTokenList } from '../selectors';
+import { hideLoadingIndication, showLoadingIndication } from '../store/actions';
+import { usePrevious } from './usePrevious';
+
+export function useAssetDetails(tokenAddress, userAddress, transactionData) {
+  const dispatch = useDispatch();
+  const tokens = useSelector(getTokens);
+  const collectibles = useSelector(getCollectibles);
+  const tokenList = useSelector(getTokenList);
+  const [currentAsset, setCurrentAsset] = useState(null);
+  const prevTokenAddress = usePrevious(tokenAddress);
+  const prevUserAddress = usePrevious(userAddress);
+  const prevTransactionData = usePrevious(transactionData);
+  useEffect(() => {
```

Comment:
We may want a new line before this `useEffect`

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/13131/commits/8c1109875b2bd628c85dd5554e316bb2b53582a5